components/state.js

```diff
@@ -0,0 +1,86 @@
+// state.js
+
+var _ = require('underscore');
+
+var Lm = require('./Lm');
+
+var inputStates = {}; 
+var outputStates = {}; 
```

Comment:
outputStates will be replaced by vni.output and vni.error, though I don't know if you intend to address that in this sprint.